<a href="https://colab.research.google.com/github/cbuitragoh/ColombianCadastreAnalysis/blob/main/Catastro.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **ANALYSIS OF THE COLOMBIAN CADASTRE**

This notebook is an EDA by the cadastre information of Colombia to 2022. The basis information get in https://geoportal.igac.gov.co/contenido/datos-abiertos-igac

In [2]:
# import necessary libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from warnings import filterwarnings

filterwarnings('ignore')

In [3]:
# loading data and displaying it

data = pd.read_csv('/content/drive/MyDrive/Catastro/TOTAL_NACIONAL_REG2.csv', encoding='latin-1')
data.head()

DEPARTAMENTO  MUNICIPIO                  NUMERO_PREDIAL  ZONA_FISICA_1  \
0           NaN        NaN                             NaN            NaN   
1           8.0     8078.0  080780001000000000001000000000            0.0   
2           8.0     8078.0  080780001000000000001000000000           40.0   
3           8.0     8078.0  080780001000000000002000000000            3.0   
4           8.0     8078.0  080780001000000000002000000000           14.0   

   ZONA_ECONOMICA_1  AREA_TERRENO_1  ZONA_FISICA_2  ZONA_ECONOMICA_2  \
0               NaN             NaN            NaN               NaN   
1               0.0            0.00            0.0               0.0   
2               8.0       210000.00            0.0               0.0   
3               6.0       163876.84            6.0               6.0   
4               6.0       120923.81           21.0               6.0   

   AREA_TERRENO_2  HABITACIONES_1  ...  AREA_CONSTRUIDA_2  HABITACIONES_3  \
0             NaN             NaN  ...                NaN             NaN   
1            0.00             0.0  ...               21.0             0.0   
2            0.00             3.0  ...               48.0             0.0   
3         4897.63             3.0  ...              200.0             0.0   
4         3220.62             0.0  ...               30.0             0.0   

   BANOS_3  LOCALES_3  PISOS_3  TIPIFICACION_3  USO_3  PUNTAJE_3  \
0      NaN        NaN      NaN             NaN    NaN        NaN   
1      0.0        0.0      0.0             0.0   20.0       60.0   
2      0.0        0.0      0.0             0.0    2.0       60.0   
3      0.0        0.0      0.0             0.0   26.0       40.0   
4      0.0        0.0      0.0             0.0    0.0        0.0   

   AREA_CONSTRUIDA_3  NUMERO_PREDIAL_ANTERIOR  
0                NaN                      NaN  
1               21.0    8078000100000001024.0  
2               71.0    8078000100000001024.0  
3              130.0    8078000100000002048.0  
4                0.0    8078000100000002048.0  

[5 rows x 34 columns]

We see that the data encoding = 'latin-1' because this data presents specific characters not available in utf-8 format. Row 0 will be removed because it is only NaN.

Here we are ready to start analyzing our data.

In [4]:
# let's see the fundamental information of the dataFrame

data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7679378 entries, 0 to 7679377
Data columns (total 34 columns):
 #   Column                   Dtype  
---  ------                   -----  
 0   DEPARTAMENTO             float64
 1   MUNICIPIO                float64
 2   NUMERO_PREDIAL           object 
 3   ZONA_FISICA_1            float64
 4   ZONA_ECONOMICA_1         float64
 5   AREA_TERRENO_1           float64
 6   ZONA_FISICA_2            float64
 7   ZONA_ECONOMICA_2         float64
 8   AREA_TERRENO_2           float64
 9   HABITACIONES_1           float64
 10  BANOS_1                  float64
 11  LOCALES_1                float64
 12  PISOS_1                  float64
 13  TIPIFICACION_1           float64
 14  USO_1                    float64
 15  PUNTAJE_1                float64
 16  AREA_CONSTRUIDA_1        float64
 17  HABITACIONES_2           float64
 18  BANOS_2                  float64
 19  LOCALES_2                float64
 20  PISOS_2                  float64
 21  TIPIFICA

it's okay. we see that our dataFrame has 2 columns of numeric values ​​(dtype float64) and 32 columns of object values. this is important to define the necessary manipulations for data processing. 

This dataFrame is memory intensive.

In [5]:
# features and characteristics of data

print('The shape of dataFrame is: ', data.shape, '\n')
print('-----------------------------------------------')
print('The unique columns in dataFrame is: ', data.columns.unique(), '\n')
print('-----------------------------------------------')
print('The total unique columns is: ', len(data.columns.unique()), '\n')
print('-----------------------------------------------')
print('The total null values in dataFrame is: ', data.isnull().sum().sum(), '\n')


The shape of dataFrame is:  (7679378, 34) 

-----------------------------------------------
The unique columns in dataFrame is:  Index(['DEPARTAMENTO', 'MUNICIPIO', 'NUMERO_PREDIAL', 'ZONA_FISICA_1',
       'ZONA_ECONOMICA_1', 'AREA_TERRENO_1', 'ZONA_FISICA_2',
       'ZONA_ECONOMICA_2', 'AREA_TERRENO_2', 'HABITACIONES_1', 'BANOS_1',
       'LOCALES_1', 'PISOS_1', 'TIPIFICACION_1', 'USO_1', 'PUNTAJE_1',
       'AREA_CONSTRUIDA_1', 'HABITACIONES_2', 'BANOS_2', 'LOCALES_2',
       'PISOS_2', 'TIPIFICACION_2', 'USO_2', 'PUNTAJE_2', 'AREA_CONSTRUIDA_2',
       'HABITACIONES_3', 'BANOS_3', 'LOCALES_3', 'PISOS_3', 'TIPIFICACION_3',
       'USO_3', 'PUNTAJE_3', 'AREA_CONSTRUIDA_3', 'NUMERO_PREDIAL_ANTERIOR'],
      dtype='object') 

-----------------------------------------------
The total unique columns is:  34 

-----------------------------------------------
The total null values in dataFrame is:  5493 




So far we know that our data frame has no duplicate columns and has 5493 null values.

let's find where these null values ​​are.

In [6]:
print('The null values by columns is:','\n', data.isnull().sum())

The null values by columns is: 
 DEPARTAMENTO                  1
MUNICIPIO                     1
NUMERO_PREDIAL                1
ZONA_FISICA_1               395
ZONA_ECONOMICA_1              1
AREA_TERRENO_1                1
ZONA_FISICA_2                44
ZONA_ECONOMICA_2              1
AREA_TERRENO_2                1
HABITACIONES_1                1
BANOS_1                       1
LOCALES_1                     1
PISOS_1                       1
TIPIFICACION_1                1
USO_1                         1
PUNTAJE_1                     1
AREA_CONSTRUIDA_1             1
HABITACIONES_2                1
BANOS_2                       1
LOCALES_2                     1
PISOS_2                       1
TIPIFICACION_2                1
USO_2                         1
PUNTAJE_2                     1
AREA_CONSTRUIDA_2             1
HABITACIONES_3                1
BANOS_3                       1
LOCALES_3                     1
PISOS_3                       1
TIPIFICACION_3                1
USO_3  

it's okay. At first glance it is detected that the null values ​​belong mostly to the column 'NUMERO_PREDIAL_ANTERIOR' and another 2 related to the 'ZONA_FISICA'

fortunately this column (NUMERO_PREDIAL_ANTERIOR) is not essential for our analyses.

In [7]:
# let's take a look at the basic statistics of the data (this function only uses numeric column values)

data.describe()

DEPARTAMENTO     MUNICIPIO  ZONA_FISICA_1  ZONA_ECONOMICA_1  \
count  7.679377e+06  7.679377e+06   7.678983e+06      7.679377e+06   
mean   4.104448e+01  4.142369e+04   2.602111e+01      8.931278e+00   
std    2.449858e+01  2.446713e+04   2.680636e+01      9.449904e+00   
min    8.000000e+00  8.078000e+03   0.000000e+00      0.000000e+00   
25%    1.800000e+01  1.875600e+04   7.000000e+00      3.000000e+00   
50%    4.100000e+01  4.155100e+04   2.000000e+01      7.000000e+00   
75%    6.600000e+01  6.644000e+04   3.100000e+01      1.200000e+01   
max    9.900000e+01  9.977300e+04   9.900000e+01      9.900000e+01   

       AREA_TERRENO_1  ZONA_FISICA_2  ZONA_ECONOMICA_2  AREA_TERRENO_2  \
count    7.679377e+06   7.679334e+06      7.679377e+06    7.679377e+06   
mean     6.988445e+04   5.875402e+00      1.727188e+00    3.110454e+04   
std      9.076317e+06   1.873377e+01      5.602028e+00    6.042107e+06   
min      0.000000e+00   0.000000e+00      0.000000e+00    0.000000e+00   
25%      7.200000e+01   0.000000e+00      0.000000e+00    0.000000e+00   
50%      1.880000e+02   0.000000e+00      0.000000e+00    0.000000e+00   
75%      4.300000e+03   0.000000e+00      0.000000e+00    0.000000e+00   
max      9.892280e+09   9.900000e+01      9.200000e+01    9.155589e+09   

       HABITACIONES_1       BANOS_1  ...     PUNTAJE_2  AREA_CONSTRUIDA_2  \
count    7.679377e+06  7.679377e+06  ...  7.679377e+06       7.679377e+06   
mean     1.456878e+00  6.158239e-01  ...  5.774259e+00       9.595339e+00   
std      1.522022e+01  7.668494e+00  ...  1.542796e+01       2.058102e+02   
min      0.000000e+00  0.000000e+00  ...  0.000000e+00       0.000000e+00   
25%      0.000000e+00  0.000000e+00  ...  0.000000e+00       0.000000e+00   
50%      0.000000e+00  0.000000e+00  ...  0.000000e+00       0.000000e+00   
75%      2.000000e+00  1.000000e+00  ...  0.000000e+00       0.000000e+00   
max      1.520000e+03  2.015000e+03  ...  9.900000e+01       3.170000e+05   

       HABITACIONES_3       BANOS_3     LOCALES_3       PISOS_3  \
count    7.679377e+06  7.679377e+06  7.679377e+06  7.679377e+06   
mean     1.217065e-02  5.704499e-03  3.591047e-03  1.471382e-02   
std      7.696545e-01  2.001081e-01  1.275634e-01  1.407748e-01   
min      0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
25%      0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
50%      0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
75%      0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
max      1.965000e+03  1.000000e+02  9.000000e+01  6.000000e+01   

       TIPIFICACION_3         USO_3     PUNTAJE_3  AREA_CONSTRUIDA_3  
count    7.679377e+06  7.679377e+06  7.679377e+06       7.679377e+06  
mean     1.314586e-02  4.689475e-01  1.554389e+00       3.473442e+00  
std      1.834840e-01  4.621174e+00  8.585530e+00       3.066969e+02  
min      0.000000e+00  0.000000e+00  0.000000e+00       0.000000e+00  
25%      0.000000e+00  0.000000e+00  0.000000e+00       0.000000e+00  
50%      0.000000e+00  0.000000e+00  0.000000e+00       0.000000e+00  
75%      0.000000e+00  0.000000e+00  0.000000e+00       0.000000e+00  
max      6.000000e+00  9.900000e+01  9.900000e+01       7.000000e+05  

[8 rows x 32 columns]

This is an interesting problem. We see that the information is displayed in a complex format to read, this is because the data types are float and object

Also keep in mind that some columns are actually categorical and not numeric. For example, 'DEPARTMENTO', 'MUNICIPIO'. Therefore, these statistics do not reveal significant information.

In [8]:
# let's see correlations between numerical variables (columns)

data.corr()

DEPARTAMENTO  MUNICIPIO  ZONA_FISICA_1  ZONA_ECONOMICA_1  \
DEPARTAMENTO           1.000000   0.999933      -0.038698         -0.103800   
MUNICIPIO              0.999933   1.000000      -0.039459         -0.106572   
ZONA_FISICA_1         -0.038698  -0.039459       1.000000          0.375434   
ZONA_ECONOMICA_1      -0.103800  -0.106572       0.375434          1.000000   
AREA_TERRENO_1         0.005834   0.005853      -0.002400         -0.001313   
ZONA_FISICA_2         -0.023771  -0.023276       0.096564          0.078424   
ZONA_ECONOMICA_2      -0.031063  -0.030982       0.031430          0.215184   
AREA_TERRENO_2         0.003951   0.003972      -0.001024          0.000441   
HABITACIONES_1        -0.016771  -0.016713       0.023473          0.022071   
BANOS_1               -0.013423  -0.013427       0.032449          0.027016   
LOCALES_1             -0.001382  -0.001462      -0.004390         -0.005886   
PISOS_1               -0.026594  -0.027678      -0.010429          0.020452   
TIPIFICACION_1        -0.020894  -0.021980       0.046233          0.052139   
USO_1                 -0.047574  -0.049140       0.144366          0.116543   
PUNTAJE_1             -0.018055  -0.019545       0.064746          0.058353   
AREA_CONSTRUIDA_1      0.001689   0.001536      -0.003366         -0.003228   
HABITACIONES_2        -0.003521  -0.003495      -0.010278         -0.002869   
BANOS_2                0.002165   0.002106      -0.023232         -0.012387   
LOCALES_2             -0.000351  -0.000381      -0.007464         -0.004871   
PISOS_2               -0.010327  -0.010439      -0.049017          0.008517   
TIPIFICACION_2        -0.007403  -0.007267      -0.037793         -0.026132   
USO_2                 -0.040648  -0.041451       0.004703          0.076212   
PUNTAJE_2              0.007849   0.007503      -0.044619          0.010591   
AREA_CONSTRUIDA_2      0.003152   0.003141      -0.007436         -0.006030   
HABITACIONES_3        -0.002078  -0.002061      -0.005311         -0.003707   
BANOS_3               -0.000307  -0.000304      -0.008913         -0.005905   
LOCALES_3             -0.002063  -0.002067      -0.008298         -0.006851   
PISOS_3               -0.008880  -0.008860      -0.026350         -0.001027   
TIPIFICACION_3        -0.007734  -0.007647      -0.021800         -0.017096   
USO_3                 -0.010447  -0.010650      -0.019099          0.009373   
PUNTAJE_3              0.006249   0.006236      -0.044654         -0.016231   
AREA_CONSTRUIDA_3      0.001095   0.001094      -0.002279         -0.001532   

                   AREA_TERRENO_1  ZONA_FISICA_2  ZONA_ECONOMICA_2  \
DEPARTAMENTO             0.005834      -0.023771         -0.031063   
MUNICIPIO                0.005853      -0.023276         -0.030982   
ZONA_FISICA_1           -0.002400       0.096564          0.031430   
ZONA_ECONOMICA_1        -0.001313       0.078424          0.215184   
AREA_TERRENO_1           1.000000       0.002057          0.003634   
ZONA_FISICA_2            0.002057       1.000000          0.728392   
ZONA_ECONOMICA_2         0.003634       0.728392          1.000000   
AREA_TERRENO_2           0.295538       0.005572          0.008283   
HABITACIONES_1          -0.000445      -0.006048         -0.007673   
BANOS_1                 -0.000503      -0.006843         -0.008317   
LOCALES_1               -0.000169       0.000387         -0.000894   
PISOS_1                 -0.004717      -0.050837         -0.066997   
TIPIFICACION_1          -0.005133      -0.063445         -0.076229   
USO_1                   -0.002142      -0.030966         -0.027090   
PUNTAJE_1               -0.004937      -0.065086         -0.077528   
AREA_CONSTRUIDA_1        0.007891       0.004434          0.001278   
HABITACIONES_2           0.000548       0.009079          0.010108   
BANOS_2                 -0.000142       0.015233          0.013815   
LOCALES_2               -0.000123       0.001662          0.000353   
PISOS_2    

Here we see that there are no significant correlations between truly numerical variables. let us remember that DEPARTMENTO and MUNICIPIO are truly categorical variables.

In [9]:
# we delete the row 0 and we verificate

data.drop([0], inplace=True)
print(data.shape)
data.head()

(7679377, 34)


DEPARTAMENTO  MUNICIPIO                  NUMERO_PREDIAL  ZONA_FISICA_1  \
1           8.0     8078.0  080780001000000000001000000000            0.0   
2           8.0     8078.0  080780001000000000001000000000           40.0   
3           8.0     8078.0  080780001000000000002000000000            3.0   
4           8.0     8078.0  080780001000000000002000000000           14.0   
5           8.0     8078.0  080780001000000000002000000000           41.0   

   ZONA_ECONOMICA_1  AREA_TERRENO_1  ZONA_FISICA_2  ZONA_ECONOMICA_2  \
1               0.0            0.00            0.0               0.0   
2               8.0       210000.00            0.0               0.0   
3               6.0       163876.84            6.0               6.0   
4               6.0       120923.81           21.0               6.0   
5               8.0       890321.03            0.0               0.0   

   AREA_TERRENO_2  HABITACIONES_1  ...  AREA_CONSTRUIDA_2  HABITACIONES_3  \
1            0.00             0.0  ...               21.0             0.0   
2            0.00             3.0  ...               48.0             0.0   
3         4897.63             3.0  ...              200.0             0.0   
4         3220.62             0.0  ...               30.0             0.0   
5            0.00             0.0  ...                0.0             0.0   

   BANOS_3  LOCALES_3  PISOS_3  TIPIFICACION_3  USO_3  PUNTAJE_3  \
1      0.0        0.0      0.0             0.0   20.0       60.0   
2      0.0        0.0      0.0             0.0    2.0       60.0   
3      0.0        0.0      0.0             0.0   26.0       40.0   
4      0.0        0.0      0.0             0.0    0.0        0.0   
5      0.0        0.0      0.0             0.0    0.0        0.0   

   AREA_CONSTRUIDA_3  NUMERO_PREDIAL_ANTERIOR  
1               21.0    8078000100000001024.0  
2               71.0    8078000100000001024.0  
3              130.0    8078000100000002048.0  
4                0.0    8078000100000002048.0  
5                0.0    8078000100000002048.0  

[5 rows x 34 columns]

let's draw the frequency curves of the numerical variables

In [ ]:
# the frequency distribution of the truly numerical variables
